### PR TITLE
Refactor our tests and code to make better use of settings.TMP_PATH

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This reverts the revert-commit 3f7fe7809343b915c4e8c6ce04aa16dad7579307.

This PR should make our ui-tests green again.

### CODE REVIEW

**Comment by diox** [comment]
This didn't fix the circleci build?

**Comment by EnTeQuAk** [comment]
> This didn't fix the circleci build?

It's a different error, circleci does have some weird permission error that I'm trying to figure out. The original error should be fixed though, I'll try hard to land this today.

File: circle.yml [hunk 1]
```diff
@@ -47,7 +47,20 @@ jobs:
...
+
+            if [ -n "$CIRCLE_TAG" ]
+            then
+              git reset --hard "$CIRCLE_SHA1"
+              git checkout -q "$CIRCLE_TAG"
+            elif [ -n "$CIRCLE_BRANCH" ]
+            then
+              git reset --hard "$CIRCLE_SHA1"
+              git checkout -q -B "$CIRCLE_BRANCH"
+            fi
```
Comment (EnTeQuAk) [inline circle.yml]: This is needed to add git clone --depth 1 support. Apparently circleci doesn't support it ootb, only version 1 did :(

File: circle.yml [hunk 2]
```diff
@@ -75,7 +88,7 @@ jobs:
             docker-compose exec web make -f Makefile-docker update_deps
             docker-compose exec web supervisorctl restart all
             # Start Test in Firefox docker container
-            docker-compose exec selenium-firefox tox -e ui-tests
+            docker-compose exec --user root selenium-firefox tox -e ui-tests
```
Comment (EnTeQuAk) [inline circle.yml]: Hack to fix the test-runs after https://github.com/jrbenny35/selenium-firefox/commit/c853be0de3b84184f9b4417be7b81e9214b5e84d

File: circle.yml (hunk 1 above)
Comment (diox) [reply circle.yml]: mmm I see that the `build-release` job doesn't have this... Worth checking with ops if we could add that there as well to speed up our build creation...

File: circle.yml (hunk 2 above)
Comment (diox) [reply circle.yml]: Is that supposed to be permanent ?

Comment (EnTeQuAk) [reply circle.yml]: definitely not, once there's an upstream fix I'll remove it

File: circle.yml (hunk 1 above)
Comment (EnTeQuAk) [reply circle.yml]: We should, I opened https://github.com/mozilla/addons-server/issues/7488 to track this. Shouldn't be much work, I'd rather do it in a separate PR though.

**Comment by jrbenny35** [comment]
Ah yes I updated my images and changed the user from 'root' to 'seluser'. That  was needed due to me using the firefox debug image now, instead of just the regular image. That ```--user root``` may have to be a permanent thing.

**Pull request merged**